src/engine/libengine/MethodFixedLayoutModern.cpp

```diff
@@ -396,5 +384,5 @@ IMCommand MethodFixedLayoutModern::handleSpecialKey(int key) {
   }
 
   // Control flow should not reach here.
-  return ret;
+  //return ret;
```

Comment:
Hmm, we have already handled every possible path but since this is a non-void function, we should at least have a return value here. Otherwise, Clang and GCC report a warning about it.

I have faced a crash #52 for not handling a possible path(logic bug).

Reply:
I was trying to figure out what happens if we don't return at all. If you handle all outcome it should not cause any problem. At least this is how it works in java. I used gcc here too but did not find any warning/error. Can You please reconfirm?

Reply:
Older GCC versions don't report a warning. I am sure that Clang reports it.

Reply:
@mominul Ok, I'll uncomment them.

Reply:
@mominul  What command do you use to build it with Clang?